client/src/components/main/RsvpListModal.tsx

```diff
@@ -16,60 +17,62 @@ import {
   TableHeader,
   TableRow,
 } from "@/components/ui/table";
+import api from "@/lib/api";
 
-export default function RsvpListModal() {
-  const attendees = [
-    {
-      firstName: "John",
-      lastName: "Doe",
-      email: "j.d@email.com",
-    },
-    {
-      firstName: "John",
-      lastName: "Doe",
-      email: "j.d@email.com",
-    },
-    {
-      firstName: "John",
-      lastName: "Doe",
-      email: "j.d@email.com",
-    },
-    {
-      firstName: "John",
-      lastName: "Doe",
-      email: "j.d@email.com",
-    },
-    {
-      firstName: "John",
-      lastName: "Doe",
-      email: "j.d@email.com",
-    },
-    {
-      firstName: "John",
-      lastName: "Doe",
-      email: "j.d@email.com",
-    },
-    {
-      firstName: "John",
-      lastName: "Doe",
-      email: "j.d@email.com",
-    },
-    {
-      firstName: "John",
-      lastName: "Doe",
-      email: "j.d@email.com",
-    },
-    {
-      firstName: "John",
-      lastName: "Doe",
-      email: "j.d@email.com",
-    },
-    {
-      firstName: "John",
-      lastName: "Doe",
-      email: "j.d@email.com",
-    },
-  ];
+//we need to connect this to the events page
+interface User {
+  first_name: string;
+  last_name: string;
+  email: string;
+}
+
+interface RSVP {
+  user: User;
+}
+
+interface RsvpListModalProps {
+  eventId: number;
+}
+
+export default function RsvpListModal({ eventId }: RsvpListModalProps) {
+  const [attendees, setAttendees] = useState<RSVP[]>([]);
+  const [eventTitle, setEventTitle] = useState<string>("");
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    async function fetchRSVPs() {
+      try {
+        const response = await api.get(`/event/${eventId}/rsvp/`);
+
+        const data = await response.data;
+        setAttendees(data);
+      } catch (error) {
+        console.error("Error fetching RSVPs:", error);
+      } finally {
+        setLoading(false);
+      }
+    }
+
+    fetchRSVPs();
+  }, [eventId]);
```

Comment:
you can use tanstack query instead. Have a look at `ping.ts` file in the hooks directory.